### ISSUE TITLE
Proof serialization/deserialization doesn't seem to be symmetric

### ISSUE THREAD

**Issue by dr-frmr** [succinctlabs/sp1#247]
Apologies if I'm missing something -- I couldn't find exactly where the deserialization is/should be occurring for the `SP1ProofWithIO` struct.

I'm attempting to share the proofs generated by the example programs, in this case the Chess example which writes `proof_with_io.json` to disk. However, when trying to deserialize the JSON to a proof, I get an error:
```
Error("invalid type: string \"01000..proof bytes...616d\", expected struct Proof", line: 1, column: 2662627)
```

I added this code inside the chess example to produce the error:
```
let proof = SP1Prover::prove(ELF, stdin).expect("proving failed");

let proof_json = serde_json::to_string(&proof).expect("serialization failed");
let mut proof = serde_json::from_str::<SP1ProofWithIO<BabyBearBlake3>>(&proof_json)
        .expect("deserialization failed");
```

Am I missing a required step, or is the `Proof` deserialization possibly missing an impl somewhere?

**Comment by puma314** [comment]
@preston-evans98 do you have insight into this? I think you last worked on serialization/deserialization of proofs and I suspect something might have gone wrong with that?

**Comment by preston-evans98** [comment]
@dr-frmr thanks for the report. Description of the issue and fix is in #249 .
